Source/WTF/wtf/cocoa/VectorCocoa.h

```diff
@@ -118,7 +119,17 @@ inline Vector<uint8_t> makeVector(NSData *data)
     return span(data);
 }
 
+template<typename T>
+inline RetainPtr<dispatch_data_t> dispatchDataFromVector(Vector<T>&& vector)
+{
+    auto vectorSize = vector.size();
+    auto vectorBuffer = vector.releaseBuffer();
+    auto* bufferPtr = vectorBuffer.span().data();
+    return adoptNS(dispatch_data_create(bufferPtr, vectorSize * sizeof(T), dispatch_get_main_queue(), makeBlockPtr([vectorBuffer = WTFMove(vectorBuffer)] { }).get()));
```

Comment:
We know that this multiplication won’t overflow, but it looks like it could, and it would be nice to have a style where it doesn’t look that way. Maybe we should use `byte_size()` instead? Can make the local variable be `bufferSpan` or `buffer` and write `buffer.data(), buffer.byte_size()`.

Reply:
Ok, going with:
```
template<typename T>
inline RetainPtr<dispatch_data_t> dispatchDataFromVector(Vector<T>&& vector)
{
    auto vectorSize = vector.size();
    auto vectorBuffer = vector.releaseBuffer();
    auto bufferSpan = vectorBuffer.span().first(vectorSize);
    return adoptNS(dispatch_data_create(bufferSpan.data(), bufferSpan.size_bytes(), dispatch_get_main_queue(), makeBlockPtr([vectorBuffer = WTFMove(vectorBuffer)] { }).get()));
}
```

Reply:
Does this mean that `releaseBuffer` creates a span with a size matching the vector _capacity_, not the vector _size_? I’m not sure we need to do it that way. If the span was the vector size, then we wouldn’t need the call to `first`.

Reply:
Yes, this is the behavior I have implemented in this PR. It is the true buffer size, which is not necessarily the same as the size of the vector (due to extra capacity). I agree though that this is a little error-prone so maybe we should indeed have have the MallocSpan size be the size of the vector.

Reply:
I think we ought to start by using the size of the vector because we don’t typically make changes to the size after releasing the buffer. If we run into any case where we want to release the buffer and then later take that additional capacity into account, we can easily make a second release function for that.